### DISCUSSION TITLE
Uninstalling pytorch-lightning

### DISCUSSION THREAD

**Discussion by acycliq** [Lightning-AI/pytorch-lightning#10942] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi

I am running `pyTorch 1.10` and `pytorch-lightning 1.5.4`. I want to downgrade pytorch-lightning to `0.7.1` because the code I am testing uses this version and It looks to me a lot of breaking changes have happened since then. 

How can I do that please? Would a `pip uninstall pytorch-lightning` and then `pip install pytorch-lightning==0.7.1` suffice or there is something else I need to take care of?

Does anyone know also if `0.7.1` is going to be compatible with  `pyTorch 1.10` ?

**Comment by awaelchli** [comment, marked as answer]
Hello
Unfortunately I must predict that your chances will be very low that 0.7.1 will run with pytorch 1.10. I would not expect it. 

But in any case, if the code was written in that version, the best is to use the pytorch version that was used in that project. Otherwise you may struggle to reproduce the results entirely. If you use conda or virtualenv, you can create different environments isolated from each other, for example, one for the old PL+pytorch project and one with the latest packages.
